arkouda/io.py

```diff
@@ -986,18 +1409,121 @@ def write_hdf5_multi_dim(
         "dset": dset,
-        "write_mode": _mode_str_to_int(mode),
+        "write_mode": mode_str_to_int(mode),
         "objType": "ArrayView",
     }
 
     generic_msg(
         cmd="tohdf",
         args=args,
     )
+
+
+def save_all(
+    columns: Union[Mapping[str, pdarray], List[pdarray]],
+    prefix_path: str,
+    names: List[str] = None,
+    file_format="HDF5",
+    mode: str = "truncate",
+    file_type: str = "distribute",
+) -> None:
+    """
+    DEPRECATED
+    Save multiple named pdarrays to HDF5 files.
+
+    Parameters
+    ----------
+    columns : dict or list of pdarrays
+        Collection of arrays to save
+    prefix_path : str
+        Directory and filename prefix for output files
+    names : list of str
+        Dataset names for the pdarrays
+    file_format : str
+        'HDF5' or 'Parquet'. Defaults to hdf5
+    mode : {'truncate' | 'append'}
+        By default, truncate (overwrite) the output files if they exist.
+        If 'append', attempt to create new dataset in existing files.
+    file_type : str ("single" | "distribute")
+            Default: distribute
+            Single writes the dataset to a single file
+            Distribute writes the dataset to a file per locale
+
+    Returns
+    -------
+    None
+
+    Raises
+    ------
+    ValueError
+        Raised if (1) the lengths of columns and values differ or (2) the mode
+        is not 'truncate' or 'append'
+
+    See Also
+    --------
+    save, load_all
+
+    Notes
+    -----
+    Creates one file per locale containing that locale's chunk of each pdarray.
+    If columns is a dictionary, the keys are used as the HDF5 dataset names.
+    Otherwise, if no names are supplied, 0-up integers are used. By default,
+    any existing files at path_prefix will be overwritten, unless the user
+    specifies the 'append' mode, in which case arkouda will attempt to add
+    <columns> as new datasets to existing files. If the wrong number of files
+    is present or dataset names already exist, a RuntimeError is raised.
+
+    Examples
+    --------
+    >>> a = ak.arange(25)
+    >>> b = ak.arange(25)
+
+    >>> # Save with mapping defining dataset names
+    >>> ak.save_all({'a': a, 'b': b}, 'path/name_prefix', file_format='Parquet')
+
+    >>> # Save using names instead of mapping
+    >>> ak.save_all([a, b], 'path/name_prefix', names=['a', 'b'], file_format='Parquet')
+    """
+    warn(
+        "ak.save_all has been deprecated. Please use ak.to_hdf or ak.to_parquet",
+        DeprecationWarning,
+    )
+    if names is not None:
+        if len(names) != len(columns):
+            raise ValueError("Number of names does not match number of columns")
+        else:
+            datasetNames = names
+    if isinstance(columns, dict):
+        pdarrays = list(columns.values())
+        if names is None:
+            datasetNames = list(columns.keys())
+    elif isinstance(columns, list):
+        pdarrays = cast(List[pdarray], columns)
```

Comment:
seems like we could use your `_bulk_write_prep` helper here

Reply:
Same thing here, deprecated function wasn't updated.